internal/engine/compiler/impl_amd64.go

```diff
@@ -3716,63 +3703,74 @@ func (c *amd64Compiler) compileCopyImpl(isTable bool, srcTableIndex, dstTableInd
 			c.assembler.CompileRegisterToRegister(amd64.ADDQ, amd64ReservedRegisterForMemory, sourceOffset.register)
 		}
 
-		beginCopyLoop := c.assembler.CompileStandAlone(amd64.NOP)
+		// handle copySize % 8 in manual mov loop
+		if !isTable {
```

Comment:
I think maybe it's time to stop having `compileCopyImpl` as it's getting hairy and complex... could you just implement compileMemoryCopy and compileMemoryTable?